internal/builder/time.go

```diff
@@ -66,39 +79,39 @@ func (b *Time) nextTime(now int64, preferred int64) int64 {
 
 func (b *Time) Queue(ctx context.Context) {
 	if !b.waiting.CompareAndSwap(false, true) {
-		b.vm.Logger().Debug("unable to acquire waiting lock")
+		b.logger.Debug("unable to acquire waiting lock")
 		return
 	}
-	preferredBlk, err := b.vm.PreferredBlock(context.TODO())
+	now := time.Now().UnixMilli()
+	preferred, gap, err := b.getPreferedTimestampAndBlockGap(now)
 	if err != nil {
-		b.waiting.Store(false)
-		b.vm.Logger().Warn("unable to load preferred block", zap.Error(err))
+		// unable to retrieve block.
```

Comment:
```suggestion
		// unable to retrieve block.
		b.waiting.Store(false)
```

should we keep `b.waiting.Store` ? This error should not happen, but seems we should keep the error handling the same with this change

Reply:
yes, good catch. I missed that.